Source/Source/Harmony/ForbidUtility_Patch.cs

```diff
@@ -66,13 +95,19 @@ public static bool Prefix(bool value)
         [HarmonyPatch(typeof(ForbidUtility), "InAllowedArea")]
         public class InAllowedArea
         {
+            // Thinktank
+            // How does vanilla do this for animals, can this be done some other way?
+            // Could maybe build a hashset of intvec3 for each pawn for a quick check?
+            // Would transpiling the original method allow for improved performance?
+
```

Comment:
Not sure how this could be improved. I was always glad it worked at all :/

Reply:
This comment was more brought out of curiousity of how vanilla handles it for animals, not due to something noticeably slow (while its not fast, I can't imagine it getting much better given the complexity of the task)